.github/workflows/build_and_test.yml

```diff
@@ -235,7 +235,8 @@ jobs:
 
       - uses: actions/setup-python@v5
         with:
-          python-version: "3.8.x"
+          # support version: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
+          python-version: "3.12.x"
```

Comment:
macos-14 only supports Python from 3.10, so let's use 3.12 for all macos versions.

Reply:
Regarding the documentation https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#using-the-python-version-input
I would write
    python-version: '3.12' instead of '3.12.x'

Reply:
sure, let me fix it.